tests/testthat/test-compare-character.R

```diff
@@ -71,3 +71,11 @@ test_that("not all lines are shown", {
   expect_equal(lines[1], "1/1 mismatches")
   expect_equal(length(lines), 8)
 })
+
+test_that("longer vectors (#513)", {
+  expect_warning(comp <- compare(letters[1:6], LETTERS[1:6]), NA)
+  expect_equal(
+    comp,
+    structure(list(equal = FALSE, message = "6/6 mismatches\nx[1]: \"a\"\ny[1]: \"A\"\n\nx[2]: \"b\"\ny[2]: \"B\"\n\nx[3]: \"c\"\ny[3]: \"C\"\n\nx[4]: \"d\"\ny[4]: \"D\"\n\nx[5]: \"e\"\ny[5]: \"E\""), class = "comparison")
```

Comment:
Upon further reflection, I'm struggling on how to do this in a more meaningful way.  Certainly I admit the test is not super clear, but part of the problem is that the original failure was because too much stuff was generated (incorrect recycling shown in blue under "new"):

<img width="246" alt="screen shot 2016-12-16 at 4 29 55 pm" src="https://cloud.githubusercontent.com/assets/6105908/21279252/5e597178-c3ad-11e6-9b90-dd1173cb6594.png">

I can write something that counts number of lines and spot checks a couple of the lines and should be easier to grasp at a glance, but this seems incomplete.  Would you be satisfied if I instead tested each component separately, but used the totality of the output of:

```
> compare(letters[1:2], LETTERS[1:2], max_diffs=1)$message
[1] "2/2 mismatches\nx[1]: \"a\"\ny[1]: \"A\""
```